dotnet/src/SemanticKernel/CoreSkills/TextMemorySkill.cs

```diff
@@ -37,49 +37,96 @@ public class TextMemorySkill
-    /// <param name="context">Contains the 'collection' to search for information and 'relevance' score</param>
-    [SKFunction("Recall a fact from the long term memory")]
+    /// <param name="text">The input text to find related memories for</param>
+    /// <param name="context">Contains the 'collection' to search for the topic and 'relevance' score</param>
+    [SKFunction("Semantic search and return up to N memories related to the input text")]
     [SKFunctionName("Recall")]
-    [SKFunctionInput(Description = "The information to retrieve")]
-    [SKFunctionContextParameter(Name = CollectionParam, Description = "Memories collection where to search for information", DefaultValue = DefaultCollection)]
+    [SKFunctionInput(Description = "The input text to find related memories for")]
+    [SKFunctionContextParameter(Name = CollectionParam, Description = "Memories collection to search", DefaultValue = DefaultCollection)]
     [SKFunctionContextParameter(Name = RelevanceParam, Description = "The relevance score, from 0.0 to 1.0, where 1.0 means perfect match",
         DefaultValue = DefaultRelevance)]
-    public async Task<string> RecallAsync(string ask, SKContext context)
+    [SKFunctionContextParameter(Name = LimitParam, Description = "The maximum number of relevant memories to recall", DefaultValue = DefaultLimit)]
+    public string Recall(string text, SKContext context)
     {
         var collection = context.Variables.ContainsKey(CollectionParam) ? context[CollectionParam] : DefaultCollection;
-        Verify.NotEmpty(collection, "Memory collection not defined");
+        Verify.NotEmpty(collection, "Memories collection not defined");
 
         var relevance = context.Variables.ContainsKey(RelevanceParam) ? context[RelevanceParam] : DefaultRelevance;
         if (string.IsNullOrWhiteSpace(relevance)) { relevance = DefaultRelevance; }
 
-        context.Log.LogTrace("Searching memory for '{0}', collection '{1}', relevance '{2}'", ask, collection, relevance);
+        var limit = context.Variables.ContainsKey(LimitParam) ? context[LimitParam] : LimitParam;
+        if (string.IsNullOrWhiteSpace(relevance)) { relevance = DefaultLimit; }
+
+        context.Log.LogTrace("Searching memories for '{0}', collection '{1}', relevance '{2}'", text, collection, relevance);
 
         // TODO: support locales, e.g. "0.7" and "0,7" must both work
-        MemoryQueryResult? memory = await context.Memory
-            .SearchAsync(collection, ask, limit: 1, minRelevanceScore: float.Parse(relevance, CultureInfo.InvariantCulture))
-            .FirstOrDefaultAsync();
+        var memories = context.Memory
+            .SearchAsync(collection, text, limit: int.Parse(limit, CultureInfo.InvariantCulture), minRelevanceScore: float.Parse(relevance, CultureInfo.InvariantCulture))
+            .ToEnumerable();
 
-        if (memory == null)
+        context.Log.LogTrace("Memories found (collection: {0})", collection);
```

Comment:
Might be worth saying how many memories we found because the number might be 0, in which case the current trace would be misleading

Reply:
Changed the wording